data-jdbc/src/main/java/io/micronaut/data/jdbc/config/SchemaGenerator.java

```
@@ -387,4 +393,31 @@ private static String resolveSql(PropertyPlaceholderResolver propertyPlaceholder
         }
         return sql;
     }
+
+    /**
+     * Filters out entities that contain properties mapped to DataType.VECTOR when the dialect is not ORACLE.
+     * This prevents schema generation errors for dialects that don't support VECTOR columns.
+     */
+    private static PersistentEntity[] filterUnsupportedVectorEntities(PersistentEntity[] entities, Dialect dialect) {
```

I think we can generate something and let the creation fail.

---

Or move classes out of tck and have them just where oracle tests are running.